docs/reference/gettingstartedexample/gse-debug-guide.md

```diff
@@ -0,0 +1,232 @@
+The cluster has the following indicators:
+- **num buffered llp packets** - Indicates the number of llp packets that are waiting to be generated.
+- **packet timing error (cycles)** - Indicates how many cycles the timestamp of the generated packet is behind expected time to generate.
+- **packet timing error** - This indicator is set if the timestamp of the generated packet was beyond the allowed packet timing error of the data.
+- **ready to generate packet** - This indicator toggles true every time the downstream data path is ready to accept a csi-2 packet.
+- **dram status** - This indicator gives you insight into the state of the DRAM manager.
+   - **bytes written** - Number of bytes written to this channel's DRAM partition.
+   - **bytes read** - Number of bytes read from this channel's DRAM partition.
+   - **overflow** - The DRAM parition could not be written to because the partition was full.
+   - **partition full** - A live status showing when the parition is full. The partition can be full, it just needs to be read from before additional writes are made.
+   - **last byte flushed** - An end of acquisition status where the last valid byte of the last packet has been read from the DRAM partition.
+- **num buffered llp timestamps** - The number of llp timestamps that are waiting to be generated. This should be close to the number of buffered llp packets.
+- **generation done** - Indicates the generation has completed.
+- **num generated packets** - The number of csi-2 packets that were sent out the serial output channel.
+
+What kinds of scenarios can you sniff out based on the indicator behavior?
+- If the generation has ended but **bytes written** is greater than **bytes read** then there is valid data getting stuck in the DRAM partition and something downstream is not ready to receive CSI-2 packets.
+- If the **overflow** indicator is true, the DRAM manager and data path is not keeping up with the incoming packets on the serial output channel. This can be caused if the host is sending more data than the DRAM manager or serial output data path can process.
+- If the **partition full** indicator is true, the partition needs to be read from. You can instrument this signal on the FPGA side to see how often the partition is getting full. It likely bounces between full and not full many times before finally overflowing. This is indicative of the host FIFO or data path not being able to generate data fast enough on the serial output channel.
+- If the **last byte flushed** stays false when a generation completes, the last packet will never have been generated.
+- If the **number of generated packets** does not match the expected number of generated packets, the data set might not be valid. This can happen if the data is expected to run faster than the serializer can output, or if the packets are getting stuck in DRAM.
+
+
+### Instrumenting and Monitoring FPGA Behavior
```

Comment:
I think maybe it's worth adding a note in this section indicating that all of these options require recompiling the FPGA bitfile, unless everyone thinks that is obvious.